src/linear_algebra/GLA.jl

```diff
@@ -156,6 +159,56 @@ plus_lm(x::AbstractVector) = begin
 end
 zero_lm(n::Int) = x::AbstractVector -> zeros(eltype(x), n)
 
+# LinearOps instance
+#-------------------
+
+@auto_hash_equals struct LinearOpDom
+  N::Int
+end
+
+@instance LinearFunctions(LinearOpDom, LinearOperator) begin
+  @import adjoint, +
+
+  dom(f::LinearOperator) = LinearOpDom(size(f,2))
+  codom(f::LinearOperator) = LinearOpDom(size(f,1))
+
+  compose(f::LinearOperator, g::LinearOperator) = g*f
+  id(V::LinearOpDom) = LOs.opEye(V.N)
+
+  oplus(V::LinearOpDom, W::LinearOpDom) = LinearOpDom(V.N + W.N)
+  oplus(f::LinearOperator, g::LinearOperator) = oplus_lo(f, g)
+  mzero(::Type{LinearOpDom}) = LinearOpDom(0)
+  braid(V::LinearOpDom, W::LinearOpDom) = braid_lo(V.N, W.N)
+  mcopy(V::LinearOpDom) =
+    LOs.opExtension(1:V.N, V.N*2)+LOs.opExtension((V.N+1):(V.N*2), V.N*2)
+  delete(V::LinearOpDom) = LOs.opZeros(0, V.N)
+  plus(V::LinearOpDom) =
+    LOs.opRestriction(1:V.N, V.N*2)+LOs.opRestriction((V.N+1):(V.N*2), V.N*2)
+  zero(V::LinearOpDom) = LOs.opZeros(V.N, 0)
+
+  plus(f::LinearOperator, g::LinearOperator) = f+g
+  scalar(V::LinearOpDom, c::Number) = LOs.opEye(typeof(c),V.N)*c
+  antipode(V::LinearOpDom) = scalar(V,-1)
+end
+
+oplus_lo(f::LinearOperator, g::LinearOperator) = begin
+  dom_total   = size(f,2) + size(g,2)
+  codom_total = size(f,1) + size(g,1)
+  dom_f   = 1:size(f,2)
+  codom_f = 1:size(f,1)
+  dom_g   = (size(f,2)+1):dom_total
+  codom_g = (size(f,1)+1):codom_total
+  fOp = LOs.opExtension(codom_f, codom_total)*f*LOs.opRestriction(dom_f,dom_total)
+  gOp = LOs.opExtension(codom_g, codom_total)*g*LOs.opRestriction(dom_g,dom_total)
+  fOp + gOp
+end
+braid_lo(v::Int, w::Int) = begin
```

Comment:
Unlike the `oplus_lo` case above, this function does need a suffix or an additional input argument. One idiomatic julia way to handle this is to pass the return type as an argument to the function. Like

```julia
function braid(::Type{LinearOperator}, v::Int, w::Int)
  upper = LOs.opExtension(1:w, v+w) * LOs.opRestriction((v+1):(v+w),v+w)
  lower = LOs.opExtension((w+1):(v+w), v+w) * LOs.opRestriction(1:v,v+w)
  upper + lower
end
```

which you would then call as `braid(LinearOperator, 5,2)`. When you pass the return type as an argument it allows multiple dispatch to find the right method of that function. This is how `zeros(Int , 16)` or `ones(Float64, 32)` know what type to fill the array with. When you do this idiom, the type gets passed as the first argument, that is just a convention they picked in the julia standard library, so people use it everywhere.

Reply:
Thanks, that's good to know! I'll get those updated.